src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -1075,6 +1088,371 @@ describe('Transactions Controller (Unit)', () => {
+        previous: null,
+        results: [moduleTransaction, multisigTransaction, incomingTransaction],
+      };
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.get.mockResolvedValueOnce({
+        data: transactionHistoryBuilder,
+      });
+      mockNetworkService.get.mockResolvedValueOnce({ data: safe });
+
+      await request(app.getHttpServer())
+        .get(`/chains/${chainId}/safes/${safeAddress}/transactions/history/`)
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.results).toHaveLength(
+            6, //3 date labels and 3 transactions
+          );
+          expect(body.results[0].timestamp).toEqual(
+            new Date('2022-12-06T00:00:00Z').getTime(),
+          );
+          expect(body.results[2].timestamp).toEqual(
+            new Date('2022-12-25T00:00:00Z').getTime(),
+          );
+          expect(body.results[4].timestamp).toEqual(
+            new Date('2022-12-31T00:00:00Z').getTime(),
+          );
+        });
+    });
+
+    it('Should change date label with time offset', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const timezoneOffset = 3600 * 2; //Offset of 2 hours
+      const chainResponse = chainBuilder().with('chainId', chainId).build();
+      const moduleTransaction = moduleTransactionTojson(
+        moduleTransactionBuilder()
+          .with('dataDecoded', null)
+          .with('executionDate', new Date('2022-12-31T22:09:36Z'))
+          .build(),
+      );
+      const safe = safeBuilder().build();
+      const transactionHistoryBuilder = {
+        count: 40,
+        next: `${chainResponse.transactionService}/api/v1/safes/${safeAddress}/all-transactions/?executed=false&limit=10&offset=10&queued=true&trusted=true`,
+        previous: null,
+        results: [moduleTransaction],
+      };
+      mockNetworkService.get.mockResolvedValueOnce({ data: chainResponse });
+      mockNetworkService.get.mockResolvedValueOnce({
+        data: transactionHistoryBuilder,
+      });
+      mockNetworkService.get.mockResolvedValueOnce({ data: safe });
+
+      await request(app.getHttpServer())
+        .get(
+          `/chains/${chainId}/safes/${safeAddress}/transactions/history/?timezone_offset=${timezoneOffset}`,
+        )
+        .expect(200)
+        .then(({ body }) => {
+          expect(body.results).toHaveLength(2);
+          expect(body.results[0].timestamp).toEqual(
+            new Date('2023-01-01T00:00:00Z').getTime(),
+          );
+        });
+    });
+
+    it('Should return correctly each transaction', async () => {
+      const chainId = faker.random.numeric();
+      const safeAddress = faker.finance.ethereumAddress();
+      const chainResponse = chainBuilder().with('chainId', chainId).build();
+      const moduleTransaction: any = getJsonResource(
+        'module-transactions/module-transaction-domain.json',
```

Comment:
> Because the purpose of this test is check that the result is correct for each field, if I use random fields we should check every field one by one, and in my opinion this is going to convert this test in a hardest test to read.

Not sure if I follow. You can check the entire payload at once instead of checking each field 🤔

Reply:
If I create random domain transactions, the response for the service should be this transactions mapped to the route entities, then I can't check the response with the builder at once, or maybe I'm missing something.